vertx-web-client/src/main/java/io/vertx/ext/web/client/impl/HttpContext.java

```diff
@@ -56,6 +56,7 @@
   private final Handler<AsyncResult<HttpResponse<T>>> handler;
   private final HttpClientImpl client;
   private final List<Handler<HttpContext<?>>> interceptors;
+  private final CacheStore privateCacheStore;
```

Comment:
I think we should not make this final and have a mutator so we don't need to expose the state and constructor to the session web client implementation